test/AWSCredentials.jl

```diff
@@ -0,0 +1,486 @@
+                @test creds.expiry > now(UTC)
+
+                # Check renew function remains unchanged
+                @test creds.renew !== nothing
+                @test creds.renew === renew
+
+                # Check force_refresh
+                creds.access_key_id = "WRONG_ACCESS_KEY"
+                creds = check_credentials(creds, force_refresh = true)
+                @test creds.access_key_id == "TEST_ACCESS_ID"
+            end
+
+            @testset "Profile" begin
+                # Check profile kwarg
+                ENV["AWS_DEFAULT_PROFILE"] = "test"
+                creds = AWSCredentials(profile="test2")
+                @test creds.access_key_id == "RIGHT_ACCESS_ID2"
+                @test creds.secret_key == "RIGHT_ACCESS_KEY2"
+
+                config = AWSConfig(profile="test2")
+                creds = config.credentials
+                @test creds.access_key_id == "RIGHT_ACCESS_ID2"
+                @test creds.secret_key == "RIGHT_ACCESS_KEY2"
+
+                # Check profile persists on renewal
+                creds.access_key_id = "WRONG_ACCESS_ID2"
+                creds.secret_key = "WRONG_ACCESS_KEY2"
+                creds = check_credentials(creds, force_refresh=true)
+
+                @test creds.access_key_id == "RIGHT_ACCESS_ID2"
+                @test creds.secret_key == "RIGHT_ACCESS_KEY2"
+            end
+
+            @testset "Assume Role" begin
+                # Check we try to assume a role
+                ENV["AWS_DEFAULT_PROFILE"] = "test:dev"
+
+                @test_ecode(
+                    "InvalidClientTokenId",
+                    AWSConfig()
+                )
+
+                # Check we try to assume a role
+                ENV["AWS_DEFAULT_PROFILE"] = "test:sub-dev"
+                let oldout = stdout
+                    r,w = redirect_stdout()
+
+                    @test_ecode(
+                        "InvalidClientTokenId",
+                        AWSConfig()
+                    )
+                    redirect_stdout(oldout)
+                    close(w)
+                    output = String(read(r))
+                    occursin("Assuming \"test:dev\"", output)
+                    occursin("Assuming \"test\"", output)
+                    close(r)
+                end
+            end
+        end
+    end
+end
+
+@testset "Retrieving AWS Credentials" begin
+    test_values = Dict{String, Any}(
+        "Default-Profile" => "default",
+        "Test-Profile" => "test",
+        "Test-Config-Profile" => "profile test",
+
+        # Default profile values, needs to match due to AWSCredentials.jl check_credentials()
+        "AccessKeyId" => "Default-Key",
+        "SecretAccessKey" => "Default-Secret",
+
+        "Test-AccessKeyId" => "Test-Key",
+        "Test-SecretAccessKey" => "Test-Secret",
+
+        "Token" => "Test-Token",
+        "InstanceProfileArn" => "Test-Arn",
+        "RoleArn" => "Test-Arn",
+        "Expiration" => now(),
```

Comment:
Should probably always use UTC right?
```suggestion
        "Expiration" => now(UTC),
```